packages/@aws-cdk/aws-certificatemanager/lambda-packages/dns_validated_certificate_handler/lib/index.js

```diff
@@ -99,15 +99,24 @@ const requestCertificate = async function(requestId, domainName, subjectAlternat
 
   console.log('Waiting for ACM to provide DNS records for validation...');
 
-  let record;
-  for (let attempt = 0; attempt < maxAttempts && !record; attempt++) {
+  let records;
+  for (let attempt = 0; attempt < maxAttempts && !records; attempt++) {
     const { Certificate } = await acm.describeCertificate({
       CertificateArn: reqCertResponse.CertificateArn
     }).promise();
     const options = Certificate.DomainValidationOptions || [];
-
     if (options.length > 0 && options[0].ResourceRecord) {
-      record = options[0].ResourceRecord;
+      // some alternative names will produce the same validation record
+      // as the main domain (eg. example.com + *.example.com)
+      // filtering duplicates to avoid errors with adding the same record
+      // to the route53 zone twice
+      const unique = options
+        .map((val) => val.ResourceRecord)
+        .reduce((acc, cur) => {
+          acc[cur.Name] = cur;
+          return acc;
+        }, {});
+      records = Object.keys(unique).sort().map((key) => unique[key]);
```

Comment:
I believe the parens around `key` are not required, so this can be: `.map(key => unique[key])`.

Reply:
You are right, they are not _needed_. Omitting them is usually considered bad coding style though, but that's probably arguable and since it's your rules here I'm happy to change it ;)